src/core/border_router/routing_manager.cpp

```diff
@@ -117,12 +118,29 @@ Error RoutingManager::SetEnabled(bool aEnabled)
 {
     Error error = kErrorNone;
 
-    VerifyOrExit(IsInitialized(), error = kErrorInvalidState);
+    switch (GetState())
+    {
+    case kStateUninitialized:
+        error = kErrorInvalidState;
+        break;
+
+    case kStateDisabled:
+        VerifyOrExit(aEnabled);
+        SetState(kStateEnabledIdle);
+        EvaluateState();
+        break;
 
-    VerifyOrExit(aEnabled != mIsEnabled);
+    case kStateEnabledIdle:
+        VerifyOrExit(!aEnabled);
+        SetState(kStateDisabled);
+        break;
 
-    mIsEnabled = aEnabled;
-    EvaluateState();
+    case kStateEnabledRunning:
+        VerifyOrExit(!aEnabled);
+        Stop();
+        SetState(kStateDisabled);
+        break;
+    }
```

Comment:
It looks like the original implementation is shorter and clearer? Are there specific reasons that we can not use the `SetState(); EvaluateState();` pattern?

Reply:
Checking the code and diff, i agree with you that it looks more complex. 

(I guess the main thing is with separate booleans we can track things differently and can have transient state e.g. we can have a case where `mIsEnabled` is false and `mIsRunning` true (disabled but still running)).

The main reason I like about using `State` is that we can log state changes. I think we may be able to log while keeping the boolean set of flags. I will close this and may push a change for the logging separately.